ee/clickhouse/views/test/test_clickhouse_retention.py

```diff
@@ -64,6 +65,66 @@ def test_retention_test_account_filters(self):
             "Day 1": {"1": ["person 3"]},
         }
 
+    @snapshot_clickhouse_queries
+    def test_retention_aggregation_by_distinct_id_and_retrieve_people(self):
+        organization = create_organization(name="test")
+        team = create_team(organization=organization)
+        user = create_user(email="test@posthog.com", password="1234", organization=organization)
+        config.AGGREGATE_BY_DISTINCT_IDS_TEAMS = f"{team.pk}"
+
+        self.client.force_login(user)
+
+        p1 = update_or_create_person(distinct_ids=["person 1", "another one"], team_id=team.pk)
+        p2 = update_or_create_person(distinct_ids=["person 2"], team_id=team.pk)
+
+        setup_user_activity_by_day(
+            daily_activity={
+                "2020-01-01": {"person 1": [{"event": "target event",}], "another one": [{"event": "target event",}],},
+                "2020-01-02": {"person 1": [{"event": "target event"}], "person 2": [{"event": "target event"}]},
+                "2020-01-03": {"another one": [{"event": "target event"}],},
+            },
+            team=team,
+        )
+
+        retention = get_retention_ok(
+            client=self.client,
+            team_id=team.pk,
+            request=RetentionRequest(
+                target_entity={"id": "target event", "type": "events"},
+                returning_entity={"id": "target event", "type": "events"},
+                date_from="2020-01-01",
+                total_intervals=3,
+                date_to="2020-01-03",
+                period="Day",
+                retention_type="retention_first_time",
+            ),
+        )
+
+        assert retention["result"][0]["values"][0]["count"] == 2
```

Comment:
I find these asserts very opaque, can we do as we are doing in the previous test?

This makes it at a glance more obvious what I'm looking at, and clear when the assertion is incorrect.

Reply:
Could do both, I need these numbers specifically because aggregating by distinctIDs, so people would return both, but I want to see one specific distinctID for one day.

Reply:
Actually, would make snapshots 2x in size, since twice going over the persons call, which is annoying.

Have instead added comments on each line, hopefully good enough?

Reply:
For feedback on these asserts, it's much clearer at least to me if it's set out like `{"cohort 1": { "interval 1":  2}, ...} ==`, it documents within the code exactly what you are expecting, as well as giving better context on failure. If we fail on one count, we see all counts around it.

Anyway, non-blocking